pkg/river/vm/vm_benchmarks_test.go

```diff
@@ -92,7 +99,7 @@ func BenchmarkExprs(b *testing.B) {
 
 			for i := 0; i < b.N; i++ {
 				vPtr := reflect.New(expectType).Interface()
-				_ = eval.Evaluate(scope, vPtr)
+				require.NoError(b, eval.Evaluate(scope, vPtr))
```

Comment:
Note that adding in require.NoError into the benchmark makes the test slower and skews the results, which is why errors were being ignored before.

Reply:
This is also what I thought, but I added the check because I doubted that it would slow down the code much. I ran a few benchmarks just now and wow, it does slow it down noticeably. 

That said, I think it wouldn't slow it down too much if we add a more lightweight check like an `err != nil` + a `b.Fatalf` and would be worth the tradeoff given that otherwise we could easily miss an error.

These are the variants which I tested:

[no_error_check.txt](https://github.com/grafana/agent/files/11706930/no_error_check.txt) (baseline):
```
_ = eval.Evaluate(scope, vPtr)
```

[require_no_error.txt](https://github.com/grafana/agent/files/11706931/require_no_error.txt):
```
require.NoError(b, eval.Evaluate(scope, vPtr))
```

[simple_error_check.txt](https://github.com/grafana/agent/files/11706932/simple_error_check.txt):
```
err = eval.Evaluate(scope, vPtr)
if err != nil {
	b.Fatalf("evaluation failed with error: %q", err)
}
```

[no_checks_even_for_parsing.txt](https://github.com/grafana/agent/files/11706929/no_checks_even_for_parsing.txt):
```
expr, _ := parser.ParseExpression(tc.input)
```
...and...
```
_ = eval.Evaluate(scope, vPtr)
```


The benchmark comparisons are:

* [no_error_check_VS_require_no_error.txt](https://github.com/grafana/agent/files/11706976/no_error_check_VS_require_no_error.txt): +45.70% geomean in sec/op

* [no_error_check_VS_simple_error_check.txt](https://github.com/grafana/agent/files/11706977/no_error_check_VS_simple_error_check.txt): +0.48% geomean in sec/op

* [no_error_check_VS_no_checks_even_for_parsing.txt](https://github.com/grafana/agent/files/11706975/no_error_check_VS_no_checks_even_for_parsing.txt): -2.48% geomean in sec/op

Based on this, IMO we could add such a simple check , and also replace the check for `parser.ParseExpression` with a simple check too. If we do both, probably it'd be a net performance increase even :) I haven't tested it though

Reply:
I'm just not sure we need our benchmarks to also be testing the code TBH, just executing it.

Reply:
For now I'm happy to just the error check I added, but if you want me to replace both with the simple checks just let me know.

Reply:
Ohh I just realised there is a `b.StopTimer()` and `b.StartTimer()`. Maybe we can just wrap the error check in them 😅

Reply:
I think that's probably overthinking it still (also stopping and starting the timer is slow too IME). Like I said, benchmarks really don't need to test that code is correct, they're just trying to measure the performance of code.